File: ICARUS/Scripts/icarus/model/EL/container/formelement/FORMELEMENT.js
/**
    @module
*/
import CONTAINER from '../CONTAINER.js';
import LABEL from '../../label/LABEL.js';
import EL from '../../EL.js';
/**
    A generic Form Element.

    @description Icarus form elements represent generic required values
    for INPUT, SELECT or TEXTAREA items. This object is
    generally passed into the Input, Select
    or TextArea to set its values.

    @class
    @extends CONTAINER
*/
export default class FORMELEMENT extends CONTAINER {
	/**
	    Constructs a Form Element 
	    @param {FORMELEMENTGROUP} node The parent
	    @param {string} element INPUT, SELECT, TEXTAREA
	    @param {MODEL} model the data model
	 */
	constructor(node, element, model) {
		model.hasSidebar = 0;
		super(node, 'DIV', model);
		this.addClass('form-element');
		//this.setClass('col-xs-12 col-sm-6 col-md-4 col-lg-offset-0');
	}
	/**
	    Construct with an associated LABEL element
	    Sets a default label of '__NoLabel' if none is provided
	*/
	construct() {
		this.label = new LABEL(this.body.pane, this.label ? this.label : this.attributes.name ? this.attributes.name : '__NoLabel');
	}
}

File: Models/Icarus/Form/FormElement.cs
using ICARUS.Models.Icarus;
using ICARUS.Models.Icarus.Elements;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Web.Mvc;
namespace ICARUS.Models {
    /// <summary>
    /// A Form Element (INPUT, SELECT etc)
    /// </summary>
    public class FORMELEMENT : Container {
        /// <summary>
        /// Generic FORMELEMENT Constructor for Entity
        /// </summary>
        public FORMELEMENT() : base("DIV", new MODEL() {
            label = "FORMELEMENT"
        }) {

        }
        /// <summary>
        /// Constructs an FORMELEMENT with the given id and label attributes already set
        /// </summary>
        /// <param name="formPost"></param>
        public FORMELEMENT(FormPost formPost) : base("FORMELEMENT", formPost) {

        }
    }
}